src/core/agent/services/connectionService.ts

```diff
@@ -83,14 +83,11 @@ class ConnectionService extends AgentService {
     });
 
     const operation = await this.resolveOobi(url, multiSigInvite);
-    const connectionId =
-      operation.done && operation.response
-        ? operation.response.i
-        : new URL(url).pathname.split("/")[2];
+    const connectionId = new URL(url).pathname.split("/")[2];
```

Comment:
Hmm, almost always the operation will be done. We are only covering an edge case. Can we keep `operation.response.i` if it's done?

Also, for the URL, can we split by "/oobi/" and then by "/". 2 is a bit of a magic number.

Reply:
`    const connectionId =
      operation.done && operation.response
        ? operation.response.i
        : new URL(url).pathname.split("/oobi/").pop();
`
Is this ok?

Reply:
@Sotatek-TungNguyen2a No, that will get the rest of the URL, which is `aid1/agent/aid2`. We need `aid1` - `new URL(url).pathname.split("/oobi/").pop().split("/")[0];`